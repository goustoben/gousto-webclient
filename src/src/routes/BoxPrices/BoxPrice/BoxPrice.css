@value ScreenMDMin from 'styles/screenSizes.css';

.container {
	max-width: 80%;
  padding: 30px 15px;
  margin: 40px auto;
  background-color: #fff;
  text-align: center;

	@media ScreenMDMin {
		max-width: 41.66666666666667%;
	}
}

.boxInfoList {
  display: flex;
	flex-direction: column;
  justify-content: space-evenly;

	@media ScreenMDMin {
		flex-direction: row;
		align-items: center;
	}

}

.title {
	composes: headerXXL from 'styles/typography.css';
	text-align: center;
}

.link {
	margin-top: 20px;
	display: flex;
	flex-direction: column;
	justify-content: space-evenly;
	align-items: center;

	@media ScreenMDMin {
		flex-direction: row;
		align-items: baseline;
	}
}


.uppercase {
	text-transform: uppercase;
}
