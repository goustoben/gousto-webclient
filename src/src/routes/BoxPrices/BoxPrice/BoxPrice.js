import PropTypes from 'prop-types'
import React, { PureComponent } from 'react'

import Image from 'Image'
import config from 'config/boxprices'
import BoxInfo from '../BoxInfo'
import { BoxPriceButton } from '../BoxPriceButton/BoxPriceButton'
import css from './BoxPrice.css'

class BoxType extends PureComponent {
  render() {
    const {
      boxInfo,
      numPersons,
      boxPricesBoxSizeSelected
    } = this.props
    const boxType = config.boxTypes[numPersons]

    return (
      <div className={css.container}>
        <h2 className={css.title}>
          {boxType.type}
          &nbsp;box
        </h2>
        <p>{boxType.description}</p>
        <Image media={boxType.image} size={0} />
        <p>Recipes in your box</p>

        <div className={css.boxInfoList}>
          {boxInfo.map((info) => (
            <BoxInfo
              key={`box-info-${numPersons}-${info.num_portions}`}
              numPortions={info.num_portions}
              pricePerPortion={info.price_per_portion}
              totalPrice={info.total}
              numPersons={numPersons}
            />
          ))}
        </div>

        <div className={css.link}>
          <p>
            Delivery is <strong className={css.uppercase}>Free</strong>
          </p>
          <BoxPriceButton
            numPersons={numPersons}
            boxPricesBoxSizeSelected={boxPricesBoxSizeSelected}
          >
            <span className={css.uppercase}>{config.cta}</span>
          </BoxPriceButton>
        </div>
      </div>
    )
  }
}

BoxType.propTypes = {
  boxInfo: PropTypes.array.isRequired,
  numPersons: PropTypes.number.isRequired,
  boxPricesBoxSizeSelected: PropTypes.func
}

BoxType.defaultProps = {
  boxPricesBoxSizeSelected: () => {}
}

export default BoxType
