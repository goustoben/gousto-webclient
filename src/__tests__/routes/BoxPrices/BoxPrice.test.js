import React from 'react'
import renderer from 'react-test-renderer';

import BoxPrice from 'routes/BoxPrices/BoxPrice'

describe("BoxPrice", () => {

	const boxPriceMock = [
		{ delivery_total: '0.00',
			gross_total: '24.99',
			num_persons: 2,
			num_portions: 2,
			price_per_portion: '6.25',
			price_per_portion_discounted: '6.25',
			product_total: '0.00',
			promo_code: 'false',
			promo_code_valid: 'false',
			recipe_discount: '0.00',
			recipe_total: '24.99',
			total: '24.99',
			vat_charged: '0.00'
		},
		{ delivery_total: '0.00',
			gross_total: '29.99',
			num_persons: 2,
			num_portions: 3,
			price_per_portion: '5.00',
			price_per_portion_discounted: '5.00',
			product_total: '0.00',
			promo_code: 'false',
			promo_code_valid: 'false',
			recipe_discount: '0.00',
			recipe_total: '29.99',
			total: '29.99',
			vat_charged: '0.00'
		},
		{ delivery_total: '0.00',
			gross_total: '34.99',
			num_persons: 2,
			num_portions: 4,
			price_per_portion: '4.37',
			price_per_portion_discounted: '4.37',
			product_total: '0.00',
			promo_code: 'false',
			promo_code_valid: 'false',
			recipe_discount: '0.00',
			recipe_total: '34.99',
			total: '34.99',
			vat_charged: '0.00'
		}
	]

	it ("Will snap BoxPrice", () => {
		const tree = renderer
		.create(<BoxPrice boxInfo={boxPriceMock} numPersons={2} />)
		.toJSON();

		expect(tree).toMatchSnapshot();
	})
})
