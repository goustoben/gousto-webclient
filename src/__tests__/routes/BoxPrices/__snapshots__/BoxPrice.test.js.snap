// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BoxPrice Will snap BoxPrice 1`] = `
<div
  className="container"
>
  <h2
    className="title"
  >
    2-Person
     box
  </h2>
  <p>
    Delicious meals for two
  </p>
  <img
    alt=""
    className="contain"
    onClick={[Function]}
    src="couples-box.jpg"
  />
  <p>
    Recipes in your box
  </p>
  <div
    className="boxInfoList"
  >
    <div
      className="container"
    >
      <p
        className="recipes"
      >
        2
         Recipes
      </p>
      <p>
        <span
          className="portionPrice"
        >
          £
          6.25
        </span>
        <small>
          per serving
        </small>
      </p>
      <p>
        <small>
          Box price
        </small>
        <span
          className="boxPrice"
        >
          £
          24.99
        </span>
      </p>
    </div>
    <div
      className="container"
    >
      <p
        className="recipes"
      >
        3
         Recipes
      </p>
      <p>
        <span
          className="portionPrice"
        >
          £
          5.00
        </span>
        <small>
          per serving
        </small>
      </p>
      <p>
        <small>
          Box price
        </small>
        <span
          className="boxPrice"
        >
          £
          29.99
        </span>
      </p>
    </div>
    <div
      className="container"
    >
      <p
        className="recipes"
      >
        4
         Recipes
      </p>
      <p>
        <span
          className="portionPrice"
        >
          £
          4.37
        </span>
        <small>
          per serving
        </small>
      </p>
      <p>
        <small>
          Box price
        </small>
        <span
          className="boxPrice"
        >
          £
          34.99
        </span>
      </p>
      <div
        className="container"
      >
        <img
          alt="Most Popular"
          className="icon"
          src="most-popular.png"
        />
      </div>
    </div>
  </div>
  <div
    className="link"
  >
    <p>
      Delivery is 
      <strong
        className="uppercase"
      >
        Free
      </strong>
    </p>
    <div
      className="container primary fill widthAuto"
    >
      <div
        className="spinnerContainer  base primary fill noHover autoWidthPadding"
        onClick={[Function]}
        onKeyUp={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      >
        <span
          className="spinner"
        >
          <div
            className="fadingCircle"
          >
            <div
              className="circle1 white"
            />
            <div
              className="circle2 white"
            />
            <div
              className="circle3 white"
            />
            <div
              className="circle4 white"
            />
            <div
              className="circle5 white"
            />
            <div
              className="circle6 white"
            />
            <div
              className="circle7 white"
            />
            <div
              className="circle8 white"
            />
            <div
              className="circle9 white"
            />
            <div
              className="circle10 white"
            />
            <div
              className="circle11 white"
            />
            <div
              className="circle12 white"
            />
          </div>
        </span>
      </div>
      <div
        className="segment base primary fill autoWidthPadding"
        onClick={[Function]}
        onKeyUp={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        role="button"
        tabIndex="0"
      >
        <a
          className="base noDecor"
          href="menu?num_portions=2"
          to="menu?num_portions=2"
        >
          <span
            className="uppercase"
          >
            Build My Box
          </span>
        </a>
      </div>
    </div>
  </div>
</div>
`;
